common/parallel/FileSystemCrawler.go

```diff
@@ -96,15 +98,25 @@ func Walk(root string, parallelism int, parallelStat bool, walkFn filepath.WalkF
 	// walk the stuff inside the root
 	reader, remainingParallelism := NewDirReader(parallelism, parallelStat)
 	defer reader.Close()
-	ctx, cancel := context.WithCancel(context.Background())
+	if appCtx != nil {
```

Comment:
For azcopy normal flow it always nil . As for azcopy there is no notion of appCtx. appCtx is valid for only in case xdatamove app. Context is created in case its nil L104.